WordPress/gutenberg
# Command Palette: Order template results in Site Editor

## What?
Order `wp_template` and `wp_template_part` entity records based on the search term in the site editor command navigation.

Fixes: https://github.com/WordPress/gutenberg/issues/51657

## Why?

This is necessary because `wp_template` and `wp_template_part` endpoints do not support `per_page` or `orderby ` parameters, so we need to sort the results based on the search term to avoid removing relevant records before they get [sliced](https://github.com/WordPress/gutenberg/pull/49330/files#diff-8e2369ac87d157769a66ee5bf038af7b5c04d397092e59afa7d33ab6ec1c5190R48).

## How?

Created a utility function which uses `.includes` to compare record titles with the command search value which is only used on results coming from `wp_template` and `wp_template_part` endpoints.

## Testing Instructions

1. Install the latest WooCommerce plugin.
2. Go to Appearance > Site Editor.
3. Click in the command navigation at the top of the Site Editor and type "Single Product".
4. The Single Product template should be returned in the results, whereas previously this was not the case.

### Testing Instructions for Keyboard




## Screenshots or screencast 

#### Before

https://github.com/WordPress/gutenberg/assets/8639742/48a49433-e8f1-4ab0-9bd0-be2605d1221f

#### After

https://github.com/WordPress/gutenberg/assets/8639742/5877a058-5d28-4fd0-9820-936cc78a941d
